backend/handlers/handlers.go

```diff
@@ -1438,6 +1439,32 @@ func (handlers *Handlers) getExchangeMoonpayBuyInfo(r *http.Request) (interface{
 	return resp, nil
 }
 
+func (handlers *Handlers) getExchangeBtcDirectInfo(r *http.Request) interface{} {
+	type result struct {
+		Success bool    `json:"success"`
+		Url     string  `json:"url"`
+		ApiKey  string  `json:"apiKey"`
+		Address *string `json:"address"`
+	}
+
+	code := accountsTypes.Code(mux.Vars(r)["code"])
+	acct, err := handlers.backend.GetAccountFromCode(code)
+	accountValid := acct != nil && acct.Offline() == nil && !acct.FatalError()
+	if err != nil || !accountValid {
+		return result{Success: false}
```

Comment:
The frontend result type also has an `errorMessage`. I'd declare it here too and use it to return "Account not valid" or smth similar.